### PR TITLE
[RUMF-1597] updates for browser billing changes

### PR DESCRIPTION
**Pull request by bcaudan**
# Motivation 

Since billing will be based on `has_replay` field, SDKs could stop sending the `_dd.session.plan` with all events.
Sampling result can still be interesting for customers if they want to know why a session wasn’t recorded.

# Changes

- deprecate `_dd.session.plan`
- add `session.sampled_for_replay` on view event

### CODE REVIEW

**Review by xgouchet — APPROVED** [review]

File: schemas/rum/view-schema.json [hunk 1]
```diff
@@ -313,6 +313,11 @@
               "description": "Whether this session is currently active. Set to false to manually stop a session",
               "default": true,
               "readOnly": true
+            },
+            "sampled_for_replay": {
+              "type": "boolean",
```
Comment (acorretti) [inline schemas/rum/view-schema.json:318]: 🥜 (nitpick) I'm fine with this or either an `"enum": [ true ]` if this value will always be true when sent

**Review by acorretti — APPROVED** [review]

File: schemas/rum/view-schema.json (hunk 1 above)
Comment (bcaudan) [reply schemas/rum/view-schema.json:318]: I had in mind to also send `false` values, it seems helpful to have `true` and `false` values in the events.
Any reason to not send them?

Comment (acorretti) [reply schemas/rum/view-schema.json:318]: It was either that model, or the `has_replay` one. Works either way!

**Review by mariusc83 — APPROVED** [review]
LGTM

**Review by detzgk — APPROVED** [review]
LGTM for the schema changes.

On the backend side, we currently default to the legacy plan if the parameter is unset. Should we be worried about this or will you continue sending this parameter until the backend changes are made?

**Comment by bcaudan** [comment]
> LGTM for the schema changes.
> 
> On the backend side, we currently default to the legacy plan if the parameter is unset. Should we be worried about this or will you continue sending this parameter until the backend changes are made?

The plan is to stop sending it with the next major release of the browser SDK, so will not start to send it before backend changes are made 🙂
I planned to ping you guys to sync about it.

**Review by kbogtob — APPROVED** [review]

**Pull request merged**